core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2337,7 +2335,10 @@ TimerTask acquisitionLockTimeoutTask() {
             return acquisitionLockTimeoutTask;
         }
 
-        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) {
+        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) throws IllegalArgumentException {
+            if (this.acquisitionLockTimeoutTask != null) {
+                throw new IllegalArgumentException("An already existing acquisition lock timeout task: " + this.acquisitionLockTimeoutTask + " is being overridden for the offset");
```

Comment:
Why to `this.acquisitionLockTimeoutTask` in error message, does it have toString method? 
```suggestion
                throw new IllegalArgumentException("Existing acquisition lock timeout exists, cannot override.");
```